device_ddf_init.cpp

```diff
@@ -556,5 +556,19 @@ bool DEV_InitDeviceBasic(Device *device)
         }
     }
 
+    zclVal.clusterId = 0x0500; // IAS Zone cluster
+    zclVal.attrId = 0x0001; // IAS Zone Type
+    zclVal.data = 0;
+
+    if (DB_LoadZclValue(&zclVal) && zclVal.data != 0)
+    {
+        ResourceItem *item = device->item(RAttrZoneType);
+        if (item && item->toNumber() != zclVal.data)
+        {
+            item->setValue(zclVal.data, ResourceItem::SourceDevice);
+            item->clearNeedPush();
+        }
+    }
+
```

Comment:
This works but you may consider just using the normal item based value storage which doesn't require extra code.

The reason I used this in the OTA version is to have the value in the database also for non DDF devices, so at the time they get a DDF the value is already available. Not sure if we need that here too.

Reply:
@manup, are you saying that any item can be used in `matchexpr`, as long as we make sure there's always one DDF that matches, so the item is created?

Reply:
After some testing, I think we do need the `RAttrZoneType` `ResourceItem` on the `Device`, but there's no need to save its value explicitly to `ZCL_VALUES`.  Still no need to have the item on the `lights` or `sensors` resource.